app/controllers/api/migration_analytics_manifest_controller.rb

```diff
@@ -0,0 +1,34 @@
+def parse_json_if_valid(json)
+    return JSON.parse(json)
+  rescue JSON::ParserError => e
+    return :invalid
+end
+
+module Api
+  class MigrationAnalyticsManifestController < BaseController
+    def index
+      tmp_manifest_path = '/tmp/migration-analytics-manifest.json'
+      tmp_manifest = File.file?(tmp_manifest_path) ? parse_json_if_valid(File.read(tmp_manifest_path)) : nil
+      tmp_manifest_valid = tmp_manifest != nil && tmp_manifest != :invalid
+
+      if tmp_manifest_valid
+        manifest_path = tmp_manifest_path
+        manifest = tmp_manifest
+      else
+        manifest_path = File.join(
+          Bundler.environment.specs['cfme-migration_analytics'].first.full_gem_path,
+          'config/default-manifest.json'
+        )
+        manifest = parse_json_if_valid(File.read(manifest_path))
+      end
+
+      res = {
+        :tmp_manifest_present => tmp_manifest != nil,
+        :tmp_manifest_valid => tmp_manifest_valid,
```

Comment:
if this is part of an API response, are these the right key names we want ? tmp_...  what if it's in a different location ?  I would expect just "manifest_present" and "manifest_valid".

Reply:
My goal here was to have the UI be able to tell if the manifest being returned is the one built-in on the appliance (the default-manifest.json file being added here) or a file they included in the /tmp/ directory, and whether the latter is being ignored because it's invalid JSON. Maybe it should be "override_manifest_present" and "override_manifest_valid" instead?

Reply:
Removed this feature for now per @agrare 's request.